src/withdraws/Withdrawer.sol

```diff
@@ -0,0 +1,39 @@
+// SPDX-License-Identifier: BSD-3-Clause
+pragma solidity ^0.8.24;
+
+import {BaseVault} from "src/BaseVault.sol";
+import {ISlisBnbStakeManager} from "src/interface/external/lista/ISlisBnbStakeManager.sol";
+import {MainnetContracts} from "script/Contracts.sol";
+import {IERC20} from "src/Common.sol";
+import {SafeERC20} from "src/Common.sol";
+
+import {BaseAsyncWithdrawStrategy} from "src/base/BaseAsyncWithdrawStrategy.sol";
+
+contract Withdrawer is BaseAsyncWithdrawStrategy {
+    /**
+     * @notice Internal function to handle the assets that are in queue for withdrawal.
+     * @param asset_ The address of the asset.
+     * @dev This function should return the amount in base denomination.
+     */
+    function _asyncWithdrawBalance(address asset_)
```

Comment:
this looks like the kind of logic that can be separated down into a separate external library 

especially as it will use multiple contracts